File: base/tps-tomcat/src/org/dogtagpki/server/tps/apdu/List_Pins_APDU.java
/* --- BEGIN COPYRIGHT BLOCK ---
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation;
 * version 2.1 of the License.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor,
 * Boston, MA  02110-1301  USA
 *
 * Copyright (C) 2007 Red Hat, Inc.
 * All rights reserved.
 * --- END COPYRIGHT BLOCK ---
 */


package org.dogtagpki.server.tps.apdu;

import org.dogtagpki.server.tps.main.TPSBuffer;

public class List_Pins_APDU extends APDU {

    public List_Pins_APDU(byte ret_size)
    {
        SetCLA((byte) 0xB0);
        SetINS((byte) 0x48);
        SetP1((byte) 0x00);
        SetP2((byte) 0x00);
        m_ret_size = ret_size;
    }

    public APDU_Type GetType()
    {
        return APDU_Type.APDU_LIST_PINS;
    }

    @Override
    public TPSBuffer GetEncoding()
    {
        TPSBuffer data = new TPSBuffer();

        data.add(m_cla);
        data.add(m_ins);
        data.add(m_p1);
        data.add(m_p2);
        data.add(m_ret_size);

        return data;
    } /* Encode */

    private byte m_ret_size = 0;

}

File: base/tools/src/main/native/tpsclient/src/apdu/List_Pins_APDU.cpp
// --- BEGIN COPYRIGHT BLOCK ---
// This library is free software; you can redistribute it and/or
// modify it under the terms of the GNU Lesser General Public
// License as published by the Free Software Foundation;
// version 2.1 of the License.
// 
// This library is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Lesser General Public License for more details.
// 
// You should have received a copy of the GNU Lesser General Public
// License along with this library; if not, write to the Free Software
// Foundation, Inc., 51 Franklin Street, Fifth Floor,
// Boston, MA  02110-1301  USA 
// 
// Copyright (C) 2007 Red Hat, Inc.
// All rights reserved.
// --- END COPYRIGHT BLOCK ---

#include <stdio.h>
#include <string.h>
#include "apdu/APDU.h"
#include "apdu/List_Pins_APDU.h"
#include "main/Memory.h"

#ifdef XP_WIN32
#define TPS_PUBLIC __declspec(dllexport)
#else /* !XP_WIN32 */
#define TPS_PUBLIC
#endif /* !XP_WIN32 */

/**
 * Constructs Set Pin APDU.
 */
TPS_PUBLIC List_Pins_APDU::List_Pins_APDU (BYTE ret_size)
{
    SetCLA(0xB0);
//    SetCLA(0x84);
    SetINS(0x48);
    SetP1(0x00);
    SetP2(0x00);
    m_ret_size = ret_size;
}

TPS_PUBLIC List_Pins_APDU::~List_Pins_APDU ()
{
}

TPS_PUBLIC APDU_Type List_Pins_APDU::GetType()
{
        return APDU_LIST_PINS;
}

TPS_PUBLIC void List_Pins_APDU::GetEncoding(Buffer &data)
{
    data += Buffer(1, m_cla);
    data += Buffer(1, m_ins);
    data += Buffer(1, m_p1);
    data += Buffer(1, m_p2);
    data += Buffer(1, m_ret_size);
} /* Encode */

